kustomize/commands/internal/util/util.go

```diff
@@ -30,19 +31,26 @@ func GlobPatterns(fSys filesys.FileSystem, patterns []string) ([]string, error)
 	return result, nil
 }
 
-// GlobPatterns accepts a slice of glob strings and returns the set of
+// GlobPatternsWithRemotes accepts a slice of glob strings and returns the set of
 // matching file paths. If files are not found, will try load from remote.
-func GlobPatternsWithLoader(fSys filesys.FileSystem, ldr ifc.Loader, patterns []string) ([]string, error) {
+func GlobPatternsWithRemotes(fSys filesys.FileSystem, patterns []string) ([]string, error) {
```

Comment:
nit: I preferred `GlobPatternsWithLoader` and to leave the function signature as is. It's more flexible if you can pass in whatever loader you want, rather than having to always use the current wd with this.

Reply:
I agree with the flexibility, but I'm concerned about encapsulation. If I only saw the name `GlobPatternsWithLoader` without reading the code, I wouldn't really know how the `ldr` is being used. 

If, as discussed above, we want the `ldr` to provide additional flexibility, say, by creating it at a different local directory that `patterns` can be relative to or at a remote url so that `patterns` can't specify that url again, we 
* are changing the spec
* risk running into contradictions. 
For example, if a pattern is found on `fSys`, but can't be loaded by `ldr` due to cyclic or other constraints, which takes precedence? We can change the spec to read something like "returns patterns found on `fSys` or `ldr`".

In such a case, though, I'd question the usefulness of the results. If I get a relative path, is it relative to my cwd, which `fSys.Glob` accounts for, or the directory that `ldr` was created for? I can't write the returned paths to a kustomization file.

On the other hand, if like the specification reads, the `ldr` is only used to load remotes, the directory that the `ldr` is created with doesn't matter. A `ldr` based at the current wd makes function behavior coherent because `fSys.Glob` checks absolute paths and the cwd, and so `ldr` wouldn't be introducing any additional constraints or finding files that `fSys.Glob` didn't intend to find. 

If I were to use this function with the `ldr` argument, I'd need to give it some thought before realizing I need to pass in a `ldr` at the cwd.

That being said, I'm not adamant about this change either. Fine with changing it back if the `ldr` is actually more beneficial!

Reply:
sure, it's a private internal package so I'm not that concerned about it, we can keep your changes in